### PR TITLE
Documentation Improvements and New Guide for 0.6.0

### PR DESCRIPTION
**Pull request by themantissa**
# Description

Improvements/fixes to the following:
* Main page, added info and links to doc for programmatic API keys #235 
* Clusters Resource - general fixes/improvements
* Network Containers and Peering - clarified purpose, examples and fixed bugs
*  Updated doc to reflect new terms:
    Cloud Provider Snapshots now Cloud Backup.
    Cloud Provider Snapshots with Point in Time Restore now Continuous Cloud Backup.
    Previous Continuous backup now Legacy Backup
* Encryption at rest, clarified only needed for BYOK.
* Teams, example usernames now email addresses
* Adds Guides section
* Adds guide for 0.6.0 upgrade needs

Link to any related issue(s): 
https://github.com/terraform-providers/terraform-provider-mongodbatlas/issues/235

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [N/A] I have signed the MongoDB CLA
- [X] I have read the Terraform contribution guidelines 
- [N/A] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [X] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code

### CODE REVIEW

**Comment by themantissa** [comment]
@gmlp this should knock out the same as https://github.com/terraform-providers/terraform-provider-mongodbatlas/pull/239.  

@shum @PacoDw @coderGo93 and @gmlp - I know this is a big PR but if you all can look over what you are most familiar with to give it a sanity check I'd appreciate it.  A lot of fixes based on helping customers and the guide is all in here.

File: website/docs/guides/0.6.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                "fail_index_key_too_long": true,
+                "javascript_enabled": true,
+                "minimum_enabled_tls_protocol": "TLS1_2",
+                "no_table_scan": false,
+                "oplog_size_mb": 0,
+                "sample_refresh_interval_bi_connector": 0,
+                "sample_size_bi_connector": 0
+              }
+            ],
+    ```
```
Comment (gmlp) [inline website/docs/guides/0.6.0-upgrade-guide.html.markdown]: Hi @themantissa , the content is great, I just suggest a change in the format, this is how is rendered: [Screen Shot 2020-06-10 at 12.21.14.png](https://trello-attachments.s3.amazonaws.com/5e46cd33162e6e10b39d259f/5ed97978c6d7fd2ebcc9bdb6/310c65ef74ee5a733070190f801670ce/Screen_Shot_2020-06-10_at_12.21.14.png)

File: website/docs/guides/0.6.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     advanced_configuration = {
+     }
+    ```
+
+    to this for 0.6.0 forward:
+
+    ```hcl
+    advanced_configuration {
+    }
+    ```
```
Comment (gmlp) [inline website/docs/guides/0.6.0-upgrade-guide.html.markdown]: same thing here: [Screen Shot 2020-06-10 at 12.24.39.png](https://trello-attachments.s3.amazonaws.com/5e46cd33162e6e10b39d259f/5ed97978c6d7fd2ebcc9bdb6/6db16bd264f99e462582e49856ee0ad7/Screen_Shot_2020-06-10_at_12.24.39.png)

**Comment by themantissa** [comment]
@gmlp latest commit thanks to you - saved me from some terrible formatting there!

**Review by gmlp — APPROVED** [review]
LGTM, great job! 👍

**Review by PacoDw — APPROVED** [review]
LGMT! Thank you so much it's a great job 👍

**Pull request merged**